clients/vault/src/oracle/agent.rs

```diff
@@ -236,9 +236,8 @@ mod tests {
 			sleep(Duration::from_secs(1)).await;
 			latest_slot = agent.last_slot_index().await;
 		}
-		// use 1 slot ahead, to ensure enough messages can be collected
-		// and to avoid "missed" messages.
-		latest_slot += 1;
+		// let's wait for envelopes and txset to be available for creating a proof
+		sleep(Duration::from_secs(5)).await;
```

Comment:
Let's hope that this finally fixes it.

Reply:
This sleep is ultimately for `poll_messages_from_stellar` [process](https://github.com/pendulum-chain/spacewalk/blob/main/clients/stellar-relay-lib/src/connection/connector/message_reader.rs#L13C21-L13C47) to have time for writing enough messages into `ScpMessageCollector`?

Reply:
Yes, to [build the proof](https://github.com/pendulum-chain/spacewalk/blob/main/clients/vault/src/oracle/agent.rs#L158-L185).